From: Modestas Vainius <modax@debian.org>
Date: Thu, 11 Oct 2018 14:24:18 +0300
Subject: Enable DLRestrictions (REJECT:OTHERSOVERSION) support for all
 libraries

Enable REJECT:OTHERSOVERSION for these libraries because their ABI is typically
not stable and structures frequently conflict.
---
 CMakeLists.txt | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 03c3a8b..83d37dd 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -359,3 +359,10 @@ install(EXPORT kdeworkspaceLibraryTargets
 
 # Add support for DebianABIManager
 include(/usr/share/pkg-kde-tools/cmake/DebianABIManager.cmake)
+
+# Enable DLResctrictions
+if (CMAKE_BUILD_TYPE STREQUAL "Debian")
+    find_package(DLRestrictions)
+    set(DEFAULT_DLRESTRICTIONS "REJECT:OTHERSOVERSION")
+    kde4deb_dlrestrictions_process_libraries()
+endif (CMAKE_BUILD_TYPE STREQUAL "Debian")
